cookbook/smart-content-data-provider.rst

```diff
@@ -42,11 +48,186 @@ How to create a custom DataProvider?
 
 To create a custom data provider you simply have to create a service which
 implements the Interface `DataProviderInterface`. This Interface provides
-functions to resolve the configured filters for the backend API with standardized
-objects and for the website with array and entity access. Additionally the
-DataProvider returns a configuration object to enable or disable features.
+functions to resolve the configured filters for the backend API with
+standardized objects and for the website with array and entity access.
+Additionally the DataProvider returns a configuration object to enable or
+disable features.
 
-For example how to implement it for ORM based entities see the Pull-Request for
-the `Account DataProvider`_.
+For ORM DataProvider there exists some abstraction layer. This layer provides the
+implementation of basic DataProvider functionality and Database query.
 
-.. _Account DataProvider: https://github.com/sulu-io/sulu/pull/1517
+If you want to create a DataProvider for the ExampleEntity you have todo
+following steps.
+
+1. Repository
+~~~~~~~~~~~~~
+
+Implement Interface `DataProviderRepositoryInterface` to provide needed function
```

Comment:
The repository has to implement the `DataProviderRepositoryInterface` and provide the `findByFilters` function.